frontend/public/components/storage/attach-storage.tsx

```diff
@@ -0,0 +1,468 @@
+/* eslint-disable no-undef, no-unused-vars */
+import * as _ from 'lodash-es';
+import * as React from 'react';
+import { Helmet } from 'react-helmet';
+import { Link } from 'react-router-dom';
+import {
+  k8sGet,
+  k8sUpdate,
+  K8sResourceKind,
+  referenceFor,
+} from '../../module/k8s';
+import * as k8sModels from '../../models';
+import {
+  ButtonBar,
+  history,
+  resourceObjPath,
+  ResourceDropdown,
+  ResourceIcon,
+} from '../utils';
+import { Checkbox } from '../checkbox';
+import { formatNamespacedRouteForResource } from '../../ui/ui-actions';
+
+const PvcDropdown = props => {
+  const kind = 'PersistentVolumeClaim';
+  const namespace = props.namespace;
+  const resources = [{ kind, namespace }];
+  const selectedKey = props.selectedKey;
+  const required = props.required;
+  return (
+    <ResourceDropdown
+      {...props}
+      desc="Storage"
+      resources={resources}
+      selectedKeyKind={kind}
+      placeholder="Select persistent volume claim"
+      selectedKey={selectedKey}
+      required={required}
+    />
+  );
+};
+
+// Helper functions for storage operations
+const createVolume = (name, claimName) => ({
+  name,
+  persistentVolumeClaim: {
+    claimName,
+  },
+});
+
+const createVolumeMount = (name, mountPath, subPath, readOnly) => {
+  const mount: any = {
+    name,
+    mountPath,
+    readOnly,
+  };
+
+  if (subPath) {
+    mount.subPath = subPath;
+  }
+
+  return mount;
+};
+
+// Gets the mount paths currently defined in the pod template.  An
+// optional filter function for matching specific containers can be used.
+const getMountPaths = (podTemplate, /* optional function */ matchContainer) => {
```

Comment:
Since we're always providing `matchContainer`, I would remove the checking that makes it optional